lib/creatubbles/base_object.rb

```diff
@@ -9,8 +9,27 @@ def self.type_name
     @type_name
   end
 
+  def self.attributes
+    @attributes || []
+  end
+
+  def attributes
+    self.class.attributes
+  end
+
+  def relationships
+    self.class.relationships
+  end
+
+  def self.relationships
+    @relationships || []
+  end
+
   def self.define_attributes(atts)
-    atts.each do |att|
+    return if @attributes
```

Comment:
You mentioned that if define_attributes will be used twice somehow, we should prevent running whole loop again. This is why I implemented that.